src/cpu/flexcpu/inflight_inst.cc

```diff
@@ -0,0 +1,648 @@
+/*
+ * Copyright (c) 2018 The Regents of The University of California
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met: redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer;
+ * redistributions in binary form must reproduce the above copyright
+ * notice, this list of conditions and the following disclaimer in the
+ * documentation and/or other materials provided with the distribution;
+ * neither the name of the copyright holders nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Authors: Bradley Wang
+ */
+
+#include "cpu/flexcpu/inflight_inst.hh"
+
+using namespace std;
+using IntReg = TheISA::IntReg;
+using PCState = TheISA::PCState;
+using FloatReg = TheISA::FloatReg;
+using FloatRegBits = TheISA::FloatRegBits;
+using MiscReg = TheISA::MiscReg;
+
+using CCReg = TheISA::CCReg;
+using VecElem = TheISA::VecElem;
+
+InflightInst::InflightInst(ThreadContext* backing_context,
+                           MemIface* backing_mem_iface,
+                           InstSeqNum seq_num,
+                           const PCState& pc_,
+                           StaticInstPtr inst_ref):
+    backingContext(backing_context),
+    backingMemoryInterface(backing_mem_iface),
+    _status(Waiting),
+    _seqNum(seq_num),
+    _pcState(pc_)
+{
```

Comment:
I don't think it should stop you from passing nullptr, since it would be totally reasonable to construct a non-memref InflightInst and not to bother giving it an implementation of this interface.